src/commands/variable-group/create.ts

```diff
@@ -0,0 +1,127 @@
+import { VariableGroup } from "azure-devops-node-api/interfaces/ReleaseInterfaces";
+import commander from "commander";
+import fs from "fs";
+import { readYaml } from "../../config";
+import { IAzureDevOpsOpts } from "../../lib/git";
+import {
+  addVariableGroup,
+  addVariableGroupWithKeyVaultMap
+} from "../../lib/pipelines/variableGroup";
+import { logger } from "../../logger";
+import { IVariableGroupData } from "../../types";
+
+/**
+ * Adds the create command to the variable-group command object
+ *
+ * @param command Commander command object to decorate
+ */
+export const createCommandDecorator = (command: commander.Command): void => {
+  command
+    .command("create")
+    .alias("c")
+    .description("Add a new variable group in Azure DevOps project.")
+    .option(
+      "-f, --file <variable-group-manifest-file-path>",
+      "Path to the yaml file that contains variable group manifest."
+    )
+    .option(
+      "-o, --org-name <organization-name>",
+      "Azure DevOps organization name; falls back to azure_devops.org in spk config"
+    )
+    .option(
+      "-p, --project <project>",
+      "Azure DevOps project name; falls back to azure_devops.project in spk config"
+    )
+    .option(
+      "-t, --personal-access-token <personal-access-token>",
+      "Personal access token associated with the Azure DevOps org; falls back to azure_devops.access_token in spk config"
+    )
+    .action(async opts => {
+      try {
+        if (!opts.file) {
+          logger.error(
+            "You need to specify a file with variable group manifest"
+          );
+          return;
+        }
+
+        const { file, orgName, project, personalAccessToken } = opts;
+
+        logger.debug(
+          `opts: ${file}, ${orgName}, ${project}, ${personalAccessToken}`
+        );
+
+        // type check
+        if (typeof orgName !== "undefined" && typeof orgName !== "string") {
+          throw Error(
+            `--org-name must be of type 'string', ${typeof orgName} specified.`
+          );
+        }
+
+        if (typeof project !== "undefined" && typeof project !== "string") {
+          throw Error(
+            `--project must be of type 'string', ${typeof project} specified.`
+          );
+        }
+
+        if (
+          typeof personalAccessToken !== "undefined" &&
+          typeof personalAccessToken !== "string"
+        ) {
+          throw Error(
+            `--personal-access-token must be of type 'string', ${typeof personalAccessToken} specified.`
```

Comment:
Shouldn't this be falling back to use personal access token from config, when this is not specified? Or, we remove the comment "falls back to azure_devops.access_token in spk config"

Same for project and orgName

Reply:
that is correct. It is only checking for the type of "string" when the argument is specified.